public/app/features/explore/TraceView/TraceView.test.tsx

```diff
@@ -5,6 +5,10 @@ import { TraceView } from './TraceView';
 import { TracePageHeader, TraceTimelineViewer } from '@jaegertracing/jaeger-ui-components';
 import { TraceSpanData, TraceData } from '@grafana/data';
 
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn(() => undefined),
```

Comment:
> Why do we need to do this?

We need to mock the useSelector hook somehow or it will throw an error that there is no Provider.

> From the code I assume this basically skips the links branch because it will return undefined instead of datasourceUID right?

You are right.

Reply:
Not sure I understand why but if we move the datalink code as per https://github.com/grafana/grafana/pull/29443#discussion_r535104287 this should go away